### ISSUE TITLE
Question on RecyclerView Integration

### ISSUE THREAD

**Issue by eicm** [bumptech/glide#493]
I was taking a look on RecyclerView integrations and ListPreloader implementation. If I haven't missed anything simply changing RecyclerToListViewScrollListener.java class to get the correct firstVisible, visibleCount and itemCount should be enough to create infinite scrolling for any other type of LayoutManager. Would you please correct me if I'm wrong.

**Comment by sjudd** [comment]
Assuming I'm understanding you, you're correct. We could definitely add an interface allowing you to pull that data out and provide implementation(s) for the default layout managers.

**Comment by eicm** [comment]
Right. I was a little bit skeptical that maybe there is a point I am missing, since only LinearLaoutManager was supported in the api.

**Comment by TWiStErRob** [comment]
It may be possible now to include those in the core library with http://b.android.com/149268.

**Comment by eicm** [comment]
What I'm asking is to have a more general interface to implement pre-loading for any extension of LayoutMangers in RecyclerViews

**Comment by sjudd** [comment]
@eicm A pull request to implement that functionality would certainly be welcome :)

**Comment by eicm** [comment]
@sjudd I can't do it this month, if it wasn't implemented in by next month I will gladly do a pull for it :)

**Comment by TWiStErRob** [comment]
What I meant is the recyclerview integration module (I didn't know it existed :) could be built-in to main library module with a `provided` dependency on support-recyclerview.

**Comment by TWiStErRob** [comment]
@eicm would you have a chance now to do a PR?

**Comment by sjudd** [comment]
This is pretty easy to do in v4, in fact the existing RecyclerView preloader is just a thin wrapper itself: https://github.com/bumptech/glide/blob/062e9ba0af80437efad8dade830737eee6c2d3c5/integration/recyclerview/src/main/java/com/bumptech/glide/integration/recyclerview/RecyclerViewPreloader.java#L30

**Comment by Kato129** [comment]
Delete file
